src/Widgets/PaymentPlan.tsx

```diff
@@ -25,23 +25,33 @@ const PaymentPlanWidget: React.FC<Props> = ({
   hideIfNotEligible,
 }) => {
   const [eligibilityPlans, status] = useFetchEligibility(purchaseAmount, apiData, configPlans)
-  const eligiblePlans = eligibilityPlans.filter((plan) => plan.eligible === true)
+  const eligiblePlans = eligibilityPlans.filter((plan) => plan.eligible)
 
-  const [isOpen, setIsOpen] = useState(false)
-  const openModal = () => setIsOpen(true)
-  const closeModal = () => setIsOpen(false)
+  const [modalProps, setModalProps] = useState({
+    isOpen: false,
+    initialPlanIndex: 0,
+  })
```

Comment:
```suggestion
  const [isOpen, setIsOpen] = useState(false)
  const [selectedPlanIndex, setSelectedPlanIndex] = useState(0)
```
Est-ce que la lecture globale ne serait pas simplifiée avec ça ?

Reply:
En effet, j'ai voulu faire ça avec une seule valeur pour effectuer une seule action lors des updates, mais vu que les setState passent en batch (au moins au onClick) ça devrait rien changer